### PR TITLE
Make sure that TransformGetItemToIndex for all graph replay

### PR DESCRIPTION
**Pull request by drisspg**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #131003
* #130871

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_higher_order_ops/flex_attention.py [hunk 1]
```diff
@@ -708,11 +708,15 @@ def sdpa_dense_backward(
         )
     grad_scores = grad_scores * scale
     grad_scores = grad_scores.to(query.dtype)
-    grad_scores = torch.where(
-        mask_graph(b, h, m, n, *mask_mod_other_buffers),
-        grad_scores,
-        torch.tensor(0, dtype=query.dtype),
+
+    mask_mod = _vmap_for_bhqkv(
```
Comment (Chillee) [inline torch/_higher_order_ops/flex_attention.py:712]: Can you add a test that would catch this?

**Review by Chillee — APPROVED** [review]

File: test/inductor/test_flex_attention.py [hunk 2]
```diff
@@ -1331,6 +1331,30 @@ def test_captured_score_mod_aot_eager_gradcheck(
             )
         )
 
+    @supported_platform
+    @common_utils.parametrize("mode", ["eager", "aot_eager"])
+    def test_document_masking_edge_case(self, mode):
```
Comment (drisspg) [inline test/inductor/test_flex_attention.py:1336]: cc @Chillee  verified this test would have caught, is that good?

Comment (Chillee) [reply test/inductor/test_flex_attention.py:1336]: oh I see, it's running in 'eager' mode. Makes sense.

**Pull request closed without merging**